miner/worker.go

```diff
@@ -395,13 +435,32 @@ func (w *worker) newWorkLoop(recommit time.Duration) {
 				signedRecent, err := p.SignRecently(w.chain, head.Block.Header())
 				if err != nil {
 					log.Info("Not allowed to propose block", "err", err)
+					//invoke pre-mining by notify txpoolsnapshotloop without blocking if not available
+					select {
+					case w.txpoolChainHeadCh <- struct{}{}:
+					default:
+					}
 					continue
 				}
 				if signedRecent {
 					log.Info("Signed recently, must wait")
+					//invoke pre-mining by notify txpoolsnapshotloop without blocking if not available
+					select {
+					case w.txpoolChainHeadCh <- struct{}{}:
+					default:
+					}
 					continue
 				}
 			}
+			//reset txpoolsnapshot and interrupt current pre-mining
```

Comment:
For 21 validators, 11 validators are not in recents, and they could reach here.
Then these 11 validators will not be allowed to do TxPoolPrefetch?